jcli/src/jcli_lib/key.rs

```diff
@@ -270,6 +273,21 @@ impl FromBytes {
     fn exec(self) -> Result<(), Error> {
         let bytes = read_hex(&self.input_bytes)?;
 
+        if self.public_key {
+            let key_bech32 = match self.key_type {
+                GenPrivKeyType::Ed25519 => bytes_to_pub_key::<Ed25519>(&bytes)?,
+                GenPrivKeyType::Ed25519Bip32 => bytes_to_pub_key::<Ed25519Bip32>(&bytes)?,
+                GenPrivKeyType::Ed25519Extended => unimplemented!(),
+                GenPrivKeyType::SumEd25519_12 => bytes_to_pub_key::<SumEd25519_12>(&bytes)?,
+                GenPrivKeyType::RistrettoGroup2HashDh => {
+                    bytes_to_pub_key::<RistrettoGroup2HashDh>(&bytes)?
+                }
+            };
+            let mut output = self.output_file.open()?;
+            writeln!(output, "{}", key_bech32)?;
+            return Ok(());
+        }
```

Comment:
Agree with @danielSanchezQ. At least, for better readability, there should be an else branch for the private key output code below.